### DISCUSSION TITLE
Case Insensitive Routing

### DISCUSSION THREAD

**Discussion by pieperz** [vikejs/vike#1304] [Discussion]
I am having some url issues due to capitalization in url routes.

The folder structure looks like pages/collections/@handle.page.tsx

For example:
https://myfavoritequiltstore.com/Collections/All returns a 404 page.
https://myfavoritequiltstore.com/collections/all returns correct page.
https://myfavoritequiltstore.com/collections/All returns correct page.

In a perfect world I would like to completely disregard capitalization on all routing.

We are hosted on vercel and I have seen a few people mention a middleware to lowercase the url.

Here is a similar discussion from next.js https://github.com/vercel/next.js/issues/21498

Is there a simple way to solve this issue with vike?

**Comment by brillout** [comment]
I created a feature request for it: https://github.com/vikejs/vike/issues/1305.

> We are hosted on vercel and I have seen a few people mention a middleware to lowercase the url.

In the meantime, would that work for you?

**Reply by brillout** [reply]
> I would like to completely disregard capitalization on all routing.

@pieperz Note that this is usually regarded as bad practice (e.g. for SEO reasons), and I believe, but feel free to disagree, that https://github.com/vikejs/vike/issues/1305 is a better appraoch.
